Новый проект

MargaritaGorshneva2-MD-25 opened the pull request:
После перехода я заново создала проект. Если необходимо-в следующий раз могу прикрепить ссылку на другой пул реквест где были прошлые версии.

MargaritaGorshneva2-MD-25 committed:
Новый проект

После перехода я заново создала проект

DolgoarshinnykhLG commented on __pycache__/config.cpython-312.pyc:
Необходимо исправить здесь и далее: эта папка лишняя в проекте. Необходимо создать .gitignore в корне проекта и добавить туда объекты не относящиеся к проекту

DolgoarshinnykhLG commented on helptest.py:
```diff
@@ -0,0 +1,28 @@
+    driver.find_element(*Locators.PASSWORD).send_keys(TEST_PASSWORD)
+    driver.find_element(*Locators.LOGIN_BUTTON_FORM).click()
+
+    WebDriverWait(driver, 10).until(EC.url_changes(LOGIN_URL))
```
Можно лучше: это либо шаги теста и должны быть размещены в тесте, либо часть предусловий и должно храниться в фикстурах (скорее второе)

DolgoarshinnykhLG commented on locators.py:
```diff
@@ -0,0 +1,53 @@
+
+class Locators:
+    # Локаторы для регистрации
+    LOGIN_ACCOUNT_BUTTON = (By.XPATH, "//div/main/section[2]/div/button[text()='Войти в аккаунт']") # Кнопка Войти в аккаунт
```
Нужно исправить здесь и далее: не стоит использовать абсолютный путь до элемента и/или его порядковый номер. Это делает локатор очень "хрупким"

DolgoarshinnykhLG commented on tests/URL.py:
```diff
@@ -0,0 +1,5 @@
+BASE_URL = "https://stellarburgers.nomoreparties.site/"
+LOGIN_URL = "https://stellarburgers.nomoreparties.site/login"
+REGIS_URL = "https://stellarburgers.nomoreparties.site/register"
+PROFILE_URL = "https://stellarburgers.nomoreparties.site/account/profile"
+FORGOT_URL = "https://stellarburgers.nomoreparties.site/forgot-password"
```
Можно лучше: адреса можно конкатенировать к BASE_URL, так будет легче их поддерживать, в том числе для нескольких стендов

DolgoarshinnykhLG commented on tests/conftest.py:
```diff
@@ -0,0 +1,21 @@
+    # chrome_options.add_argument("--headless") # раскомментируйте для headless режима
+
+    current_directory = os.path.dirname(os.path.abspath(__file__))  # Исправлено: __file__
+    driver_path = os.path.join(current_directory, "chromedriver.exe")
```
Можно лучше: в целом, решение рабочее и даже вполне переносимое (если версия установленного браузера совместима, по-хорошему её тоже в README стоит тогда указывать), но в случае изменения версии браузера (а адаптер под браузер пишется строго под конкретную его версию) нужно будет снова коммитить в его репозиторий. Идеальным решением считается прописать путь драйвера в PATHS, он будет автоматически подхватывать путь в фикстуре

DolgoarshinnykhLG commented on tests/test_constructor_navigation.py:
```diff
@@ -0,0 +1,45 @@
+
+    WebDriverWait(driver, 10).until(
+        EC.visibility_of_element_located(Locators.FILLINGS_MENU)
+    )
```
Необходимо исправить здесь и далее: Каждый тест должен заканчиваться проверкой в виде assert-инструкции

DolgoarshinnykhLG commented on tests/test_exampl.py:
```diff
@@ -0,0 +1,43 @@
+        EC.element_to_be_clickable(Locators.CONSTRUCTOR)
+    )
+    constructor_button.click()
+    time.sleep(2)  # Пауза в 2 секунды
```
Необходимо исправить здесь и далее: использование sleep не допускается. Стоит посмотреть в сторону нативных expected conditions в selenium и подобрать подходящее. Ниже оно уже чуть есть. Возможно, стоит просто чуть увеличить время ожидания?

DolgoarshinnykhLG commented on tests/test_registration.py:
```diff
@@ -0,0 +1,40 @@
+class TestRegistrationPage:
+    def test_successful_registration(self, driver):
+        random_email = generate_random_email()
+        print(f"Используемый email: {random_email}") # Вывод email для отладки
```
Нужно исправить: от любых принтов в проекте лучше избавиться. Если есть желание логировать что-то, то лучше использовать пакет logging

DolgoarshinnykhLG commented on tests/test_registration.py:
```diff
@@ -0,0 +1,40 @@
+        WebDriverWait(driver, 10).until(EC.element_to_be_clickable(Locators.REGISTER_BUTTON))
+        driver.find_element(*Locators.REGISTER_BUTTON).click()
+
+        # Здесь нужно добавить проверку на сообщение об ошибке или другое поведение при неудачной регистрации
```
Необходимо исправить: выглядит как забытая ту-душка

MargaritaGorshneva2-MD-25 committed:
Доработка

В локаторах у емейл пришлось оставить длинное так как он иначе не видит элемент

MargaritaGorshneva2-MD-25 committed:
Забыла добавить файл

MargaritaGorshneva2-MD-25 committed:
Неправильно добавила файл

DolgoarshinnykhLG commented on __pycache__/helptest.cpython-312.pyc:
Можно лучше: в gitignore директория добавлена, но нужно будет ещё удалить эту директорию и закоммитить изменения, иначе папка останется в gitignore, но всё равно будет влита в целевую ветку, т.к. была закоммичена раньше
